.github/workflows/pull-request-podman.yml

```diff
@@ -0,0 +1,69 @@
+      - 'components/container_image.py'
+      - 'directord/components/lib/podman.py'
+      - 'orchestrations/functional-podman-test.yaml'
+
+jobs:
+  functional_podman_functional:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+      - name: Run install
+        run: sudo bash tools/dev-setup.sh /opt/directord python3 ${{ github.workspace }}
+      - name: Run server service install
+        run: |
+          sudo /opt/directord/bin/directord-server-systemd
+          sudo systemctl daemon-reload
+          sudo systemctl restart directord-server
+      - name: Build current container and run client
+        run: |
+          sudo mkdir /tmp/containers
+          sudo apt-get install -y podman
+          sudo podman build -t test-image .
+          sudo podman run --hostname container-client \
+          --name directord-client --net=host \
+          --env DIRECTORD_ZMQ_SERVER_ADDRESS=127.0.0.1 \
+          --env DIRECTORD_DEBUG=True --user 0 --detach  \
+          -v /tmp/containers:/var/lib/containers:Z  \
+          --privileged test-image directord --debug
+          sudo podman exec directord-client microdnf install -y podman
+          sudo podman exec directord-client podman system service --log-level debug -t 0 unix:///tmp/podman.sock &
+          sudo podman run --rm --name registry -d -p 5000:5000 registry:2
+
+      - name: Wait for client online
+        run: |
+          timeout 120 bash -c 'while ! sudo /opt/directord/bin/directord manage --list-nodes; do sleep 1; done'
+      - name: Execute functional check
+        run: |
+          cd /opt/directord/share/directord/orchestrations
+          sudo timeout 240 /opt/directord/bin/directord \
+                                              orchestrate \
+                                              functional-podman-test.yaml \
+                                              --poll \
+                                              --check
+
+      - name: Generate log details
+        run: |
+          sudo podman logs directord-client &> /tmp/directord-client-podman.log || true
+          sudo journalctl -u directord-client -n 2000 &> /tmp/directord-client.log || true
+          sudo journalctl -u directord-server -n 2000 &> /tmp/directord-server.log || true
+          sudo /opt/directord/bin/directord manage --list-nodes &> /tmp/directord-nodes.log
+          sudo /opt/directord/bin/directord manage --dump-cache &> /tmp/directord-cache.log
+          sudo /opt/directord/bin/directord manage --export-jobs /tmp/directord-jobs-export.log
+        if: failure()
+      - name: Upload build Log artifacts on failure
+        if: failure()
+        uses: actions/upload-artifact@v2
+        with:
+          name: functional-realworld-test-logs
```

Comment:
The name of this should match the job name `functional-podman-functional`.

Reply:
ack